avocado/core/task/runtime.py

```diff
@@ -46,3 +55,171 @@ def __eq__(self, other):
+        # creates the tasks and runtime tasks for the requirements
+        for requirement_runnable in requirements_runnables:
+            name = '%s-%s' % (requirement_runnable.kind,
+                              requirement_runnable.kwargs.get('name'))
+            prefix = '%s-%s' % (test_suite_name, name)
+            # the human UI works with TestID objects, so we need to
+            # use it to name Task
+            task_id = TestID(prefix, name)
+            # with --dry-run we don't want to run requirement
+            if runnable.kind == 'dry-run':
+                requirement_runnable.kind = 'noop'
+            # creates the requirement task
+            requirement_task = Task(requirement_runnable,
+                                    identifier=task_id,
+                                    status_uris=status_server_uri,
+                                    category='requirement',
+                                    job_id=job_id)
+            requirements_runtime_tasks.append(cls(requirement_task))
+
+        return requirements_runtime_tasks
+
+
+class RuntimeTaskGraph:
+    """Graph representing dependencies between runtime tasks."""
+
+    def __init__(self, tests, test_suite_name, status_server_uri, job_id):
+        """Instantiates a new RuntimeTaskGraph.
+
+        From the list of tests, it will create runtime tasks and connects them
+        inside the graph by its dependencies.
+
+        :param tests: variants of runnables from test suite
+        :type tests: list
+        :param test_suite_name: test suite name which this test is related to
+        :type test_suite_name: str
+        :param status_server_uri: the URIs for the status servers that this
+                                  task should send updates to.
+        :type status_server_uri: list
+        :param job_id: the ID of the job, for authenticating messages that get
+                       sent to the destination job's status server and will
+                       make into the job's results.
+        :type job_id: str
+        """
+        self.graph = {}
+        # create graph
+        no_digits = len(str(len(tests)))
+        for index, (runnable, variant) in enumerate(tests, start=1):
+            runnable = deepcopy(runnable)
+            runtime_test = RuntimeTask.get_test_from_runnable(
+                runnable,
+                no_digits,
+                index,
+                variant,
+                test_suite_name,
+                status_server_uri,
+                job_id)
+            self.graph[runtime_test] = runtime_test
+
+            requirements_tasks = RuntimeTask.get_requirements_form_runnable(
+                runnable,
+                test_suite_name,
+                status_server_uri,
+                job_id)
+            self._connect_requirments_with_test(requirements_tasks,
+                                                runtime_test)
+
+    def _add_edge(self, pre_runtime_task, post_runtime_task):
```

Comment:
Do you think a single liner like this is worthy to be a method?  IMO, it isn't, but if you disagree, please make it a static method.